setup.py

```diff
@@ -1,15 +1,20 @@
 import os
 import subprocess
 import warnings
-from packaging import version
-import pip
+from setuptools._vendor.packaging import version
 
-if version.parse(pip.__version__) < version.parse("19.0"):
-    # version 19.0 is required, since we are using pyproject.toml for definition of build-time depdendencies. See https://pip.pypa.io/en/stable/news/#id209
-    warnings.warn("You are using pip version {}. However, pip version > 19.0 is recommended. You can continue with the installation, but installation problems can occour. Please refer to https://github.com/precice/python-bindings#build-time-dependencies-cython-numpy-defined-in-pyprojecttoml-are-not-installed-automatically for help.".format(pip.__version__))
+# If installed with pip we need to check its version
+try:
+    import pip
+    if version.parse(pip.__version__) < version.parse("19.0"):
+        # version 19.0 is required, since we are using pyproject.toml for definition of build-time depdendencies. See https://pip.pypa.io/en/stable/news/#id209
+        warnings.warn("You are using pip version {}. However, pip version > 19.0 is recommended. You can continue with the installation, but installation problems can occour. Please refer to https://github.com/precice/python-bindings#build-time-dependencies-cython-numpy-defined-in-pyprojecttoml-are-not-installed-automatically for help.".format(pip.__version__))
+
+    if version.parse(pip.__version__) < version.parse("10.0.1"):
+        warnings.warn("You are using pip version {}. However, pip version > 10.0.1 is required. If you continue with installation it is likely that you will face an error. See https://github.com/precice/python-bindings#version-of-pip3-is-too-old".format(pip.__version__))
+except:
+    warnings.warn("Assuming that you are not using pip!")
```

Comment:
I think this is a valid use-case (also outside of the "spack world"), since a user might just run `python3 setup.py install`. Then `pip` is not required. As far as I know running `python3 setup.py install` bypasses the dependencies defined in `pyproject.toml`. I would not fix this by duplicating the content of `pyproject.toml`, since the recommended approach is to use `pip`.

We should instead throw a warning, if `pip` is not used:

```suggestion
try:
    import pip
    if version.parse(pip.__version__) < version.parse("19.0"):
        # version 19.0 is required, since we are using pyproject.toml for definition of build-time depdendencies. See https://pip.pypa.io/en/stable/news/#id209
        warnings.warn("You are using pip version {}. However, pip version > 19.0 is recommended. You can continue with the installation, but installation problems can occour. Please refer to https://github.com/precice/python-bindings#build-time-dependencies-cython-numpy-defined-in-pyprojecttoml-are-not-installed-automatically for help.".format(pip.__version__))

    if version.parse(pip.__version__) < version.parse("10.0.1"):
        warnings.warn("You are using pip version {}. However, pip version > 10.0.1 is required. If you continue with installation it is likely that you will face an error. See https://github.com/precice/python-bindings#version-of-pip3-is-too-old".format(pip.__version__))
except:
    warnings.warn("Assuming that you are not using pip! If the bindings work correctly, you do not have to do anything. If you face problems during installation or running pyprecice, this means that you have to make sure that all dependencies are installed correctly and repeat the installation of pyprecice. Refer to pyproject.toml for a list of dependencies.")
```

Reply:
This has been incorporated in b72f5cc186f5889702232315905fda79d14e3a5a